app/src/main/java/com/hemdan/mvvm/ui/main/actorslist/ActorsListFragment.kt

```diff
@@ -0,0 +1,62 @@
+package com.hemdan.mvvm.ui.main.actorslist
+
+import android.os.Bundle
+import android.view.View
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.hemdan.mvvm.R
+import com.hemdan.mvvm.data.model.PopularInfo
+import com.hemdan.mvvm.ui.base.BaseFragment
+import kotlinx.android.synthetic.main.fragment_actors_list.*
+import java.util.ArrayList
+import javax.inject.Inject
+
+class ActorsListFragment : BaseFragment() {
+
+    private lateinit var listAdapter: ActorsAdapter
+    private var listLayoutManager: LinearLayoutManager? = null
+
+    private var popularInfos = ArrayList<PopularInfo>()
+
+    override fun getLayoutResourceId(): Int = R.layout.fragment_actors_list
+
+    @Inject
+    lateinit var actorsListViewModel: ActorsListViewModel
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        listLayoutManager = LinearLayoutManager(context)
+        configRecycleView(popularInfos)
+        observeViewModel()
+        actorsListViewModel.getActors()
+
+        actorsList.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+            override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                super.onScrolled(recyclerView, dx, dy)
+                val currentItems = listLayoutManager!!.childCount
+                val scrolledItems = listLayoutManager!!.findFirstCompletelyVisibleItemPosition()
```

Comment:
@MonaAliMahmoud  as recommended don't use !! you can use ?.let instead or if used the setup method I proposed above listActors.getLayoutManager.findFirstCompleteVis....etc